app/lib/shared/integrity.dart

```diff
@@ -19,296 +19,266 @@ final _logger = Logger('integrity.check');
 class IntegrityChecker {
   final DatastoreDB _db;
   final int _concurrency;
-  final _problems = <String>[];
-
-  final _userToOauth = <String?, String?>{};
-  final _oauthToUser = <String?, String?>{};
-  final _emailToUser = <String?, List<String?>>{};
-  final _deletedUsers = <String?>{};
-  final _invalidUsers = <String?>{};
-  final _userToLikes = <String?, List<String?>>{};
-  final _packages = <String?>{};
-  final _packageReplacedBys = <String?, String?>{};
-  final _packagesWithVersion = <String?>{};
-  final _moderatedPackages = <String?>{};
-  final _publishers = <String?>{};
-  final _publishersAbandoned = <String?>{};
+
+  final _userToOauth = <String, String?>{};
+  final _oauthToUser = <String, String>{};
+  final _deletedUsers = <String>{};
+  final _invalidUsers = <String>{};
+  final _packages = <String>{};
+  final _packageReplacedBys = <String, String>{};
+  final _packagesWithVersion = <String>{};
+  final _publishers = <String>{};
+  final _publishersAbandoned = <String>{};
   int _packageChecked = 0;
   int _versionChecked = 0;
 
   IntegrityChecker(this._db, {int? concurrency})
       : _concurrency = concurrency ?? 1;
 
   /// Runs integrity checks, and reports the list of problems.
-  Future<List<String>> check() async {
-    await _checkUsers();
-    await _checkOAuthUserIDs();
-    await _checkPublishers();
-    await _checkPublisherMembers();
-    await _checkPackages();
-    await _checkVersions();
-    await _checkLikes();
-    await _checkModeratedPackages();
-    return _problems;
+  Stream<String> check() async* {
```

Comment:
Would it make any sense to do some of these in parallel?

Reply:
In some case it may be good, but others rely on cached values from earlier methods.